feat: youtube to anchor workflow (#805)
asyncapi/community: MEETINGS_ORGANIZATION.md

#### How to find YouTube ID?

Let's take this video link, for example: https://www.youtube.com/watch?v=3rg_7hIb9PQ.

Here the video ID is the word/entity after `https://www.youtube.com/watch?v=`, i.e., `3rg_7hIb9PQ`

### List of different types of YouTube links

- `https://www.youtube.com/watch?v=VIDEO_ID`
  - Here, it can be https://www.youtube.com/watch?v=3rg_7hIb9PQ
- `https://youtu.be/VIDEO_ID`
  - Here, it can be https://www.youtu.be/3rg_7hIb9PQ
- `https://www.youtube.com/embed/VIDEO_ID`
  - Here, it can be https://www.youtube.com/embed/3rg_7hIb9PQ
- `https://www.youtube.com/playlist?list=PLAYLIST_ID`
  - Here, it can be https://www.youtube.com/playlist?list=PLbi1gRlP7piiaD67o1F4EOPoZztg2r8l6
- `https://www.youtube.com/shorts/VIDEO_ID`
  - Here, it can be https://www.youtube.com/shorts/3rg_7hIb9PQ
- `https://www.youtube.com/watch?v=VIDEO_ID&list=PLAYLIST_ID`
  - Here, it can be https://www.youtube.com/watch?v=deLUAobdVpw&list=PLbi1gRlP7piiaD67o1F4EOPoZztg2r8l6
- `https://youtube.com/shorts/VIDEO_ID?feature=share`
  - Here, it can be https://youtube.com/shorts/U5jUr8XAF_M?feature=share

> ### Step-by-step procedure
> 
> Here is an example with a procedure on how to use this workflow:
> 
> 1. Go to the `Actions` section on GitHub.
> 2. Scroll down the menu on the left-hand side.
> 3. Click on the `Upload Episode from YouTube To Spotify for Podcasters` option.
> 4. Click on the `Run Workflow` button.
> 5. Provide the YouTube ID in the box.
> 6. Click on `Run workflow` button. 
> 7. Wait for the workflow to finish. (Depending on video size, it may take some time to upload, so please have patience. Thank you.) 
> 8. Go to podcasters.spotify.com (Login to your account if you haven't already).
> 9. Go to `Episodes` section. There, you will see your latest upload.
> 10. Hurray! Your episode is now uploaded successfully!

Reviewer: ```suggestion
### Step-by-step procedure

Here is an example with a procedure on how to use this workflow:

1. Go to the `Actions` section on GitHub.
2. Scroll down the menu on the left-hand side.
3. Click on the `Upload Episode from YouTube To Spotify for Podcasters` option.
4. Click on the `Run Workflow` button.
5. Provide the YouTube ID in the box.
6. Click on `Run workflow` button. 
7. Wait for the workflow to finish. (Depending on video size, it may take some time to upload, so please have patience. Thank you.) .
8. A few moments later....Hurray! Your episode is now uploaded successfully!

You can also modify description and other information for uploaded podcast directly in Spotify:

1. Go to https://podcasters.spotify.com/.
2. Login using AsyncAPI account.
2. Go to `Episodes` section. There, you will see your latest upload.
3. Perform edits and save them.
```

how about like this?

Author: This looks good! Thank you for your suggestion! :rocket: :+1: 
cc: @thulieblack

Author: applied suggestion through eaa2363 :100:

- `https://www.youtube.com/playlist?list=PLAYLIST_ID`
  - Here, it can be https://www.youtube.com/playlist?list=PLbi1gRlP7piiaD67o1F4EOPoZztg2r8l6
- `https://www.youtube.com/shorts/VIDEO_ID`
  - Here, it can be https://www.youtube.com/shorts/3rg_7hIb9PQ
- `https://www.youtube.com/watch?v=VIDEO_ID&list=PLAYLIST_ID`
  - Here, it can be https://www.youtube.com/watch?v=deLUAobdVpw&list=PLbi1gRlP7piiaD67o1F4EOPoZztg2r8l6
- `https://youtube.com/shorts/VIDEO_ID?feature=share`
  - Here, it can be https://youtube.com/shorts/U5jUr8XAF_M?feature=share

***

### Step-by-step procedure

Here is an example with procedure on how to use this workflow:

1. Go to the `Actions` section on GitHub.
2. Scroll down the menu on the left hand-side.
3. Click on the `Upload Episode from YouTube To Spotify for Podcasters` option.
4. Click on the `Run Workflow` button.
5. Provide the YouTube ID in the box.
6. Click on `Run workflow` button. 
7. Wait for the workflow to finish. (May take some time to upload depending upon video size so please have patience. Thank you.) 
8. Go to podcasters.spotify.com (Log in to your account if you haven't already).

> 9. Go to `Episodes` section. There you will see your latest upload as `Draft` 

Reviewer: I guess we can remove draft stuff from here now

Author: resolved with 9cf4d4d25fdc1fc3bc8a519acb6d903715234eeb

### How can I become a host?

Once your `Regular` meeting gets approved, contact `Fran Mendez` so that he can give you access to all required tools.

## How do I publish broadcasted meeting as podcast?

### Why do we need this feature?

- There are some people who favor listening the video/livestream as an audio (podcast) instead of watching the video.

> - The goal is to upload our AsyncAPI video/livestreams on `Spotify For Podcasters` platform which will ultimately allow the listener to play the episodes on Spotify. Almost every YouTube video can be converted into a podcast.

Reviewer: Almost? are there some videos that cannot be converted?

Author: @derberg 
- I didn't tried the workflow on 'unlisted' and 'private' videos.
- I suppose 'private' ones are private for a reason and should not be uploaded.
- 'unlisted' ones are like if you're having some content that should be visible to some specific people as such. Not sure 'unlisted' are to be considered here.

Reviewer: private or unlisted should anyway not go to spotify, so yeah, just ignore these

Author: Okay. Understood

> - This workflow provides the solution to the above problem.

Reviewer: ```suggestion
```

not relevant info for FAQ

### Who can upload the videos?

- The workflow can only be triggered by a person who has the write-access to the `master` branch of `asyncapi/community` repository.

### What is needed?

- The YouTube video ID is needed to trigger the workflow.

### What is YouTube video ID?

- A YouTube video ID is a unique combination of characters that identifies a specific video on the YouTube platform. It's used in the URL of the video to direct users to the exact video they want to watch. The video ID is typically a sequence of letters, numbers, and special characters that generally comes after the "v=" parameter in the URL.
- [More examples](#list-of-different-types-of-youtube-links) of video ID are given below: ⬇️

#### How to find YouTube ID?

Let's take this video link for example: https://www.youtube.com/watch?v=3rg_7hIb9PQ

Here the video ID is the word/entity after `https://www.youtube.com/watch?v=`, i.e., `3rg_7hIb9PQ`

> ***

Reviewer: ?

Author: I tried explaining what is a YouTube video ID. Is it confusing in any way, or maybe, difficult to understand?

Reviewer: no, confusing is the `***` there

Author: It adds a divider in the document 
More info (here)[https://www.markdownguide.org/basic-syntax/#horizontal-rules]

Reviewer: yeah, markdown thingy, but why we need it, we have sections for that

Author: Added it thinking about better readability. But if it looks odd or doesn't matches the standard, I can remove it.
Please let me know about this @derberg

Reviewer: better remove as we do not have it anywhere else

Author: Got it.🫡

Author: resolved with 9e0e608

### List of different types of YouTube links

- `https://www.youtube.com/watch?v=VIDEO_ID`
  - Here, it can be https://www.youtube.com/watch?v=3rg_7hIb9PQ
- `https://youtu.be/VIDEO_ID`
  - Here, it can be https://www.youtu.be/3rg_7hIb9PQ
- `https://www.youtube.com/embed/VIDEO_ID`
  - Here, it can be https://www.youtube.com/embed/3rg_7hIb9PQ
- `https://www.youtube.com/playlist?list=PLAYLIST_ID`
  - Here, it can be https://www.youtube.com/playlist?list=PLbi1gRlP7piiaD67o1F4EOPoZztg2r8l6
- `https://www.youtube.com/shorts/VIDEO_ID`
  - Here, it can be https://www.youtube.com/shorts/3rg_7hIb9PQ
- `https://www.youtube.com/watch?v=VIDEO_ID&list=PLAYLIST_ID`
  - Here, it can be https://www.youtube.com/watch?v=deLUAobdVpw&list=PLbi1gRlP7piiaD67o1F4EOPoZztg2r8l6
- `https://youtube.com/shorts/VIDEO_ID?feature=share`
  - Here, it can be https://youtube.com/shorts/U5jUr8XAF_M?feature=share

> *If you find more formats, please add to this list by raising an issue and with a follow-up Pull Request (PR)*

Reviewer: ```suggestion
```

lets avoid such sentences, PR can be opened for any part of this document to improve it

Author: Understood ![salute-emoji](https://github.com/asyncapi/community/assets/99868037/52cedc17-1f62-441b-a3bc-9cf5967cdf91)

***

### Step-by-step procedure

Here is an example with procedure on how to use this workflow:

1. Go to the `Actions` section on GitHub.
2. Scroll down the menu on the left hand-side.
3. Click on the `Upload Episode from YouTube To Spotify for Podcasters` option.
4. Click on the `Run Workflow` button.
5. Provide the YouTube ID in the box.
6. Click on `Run workflow` button. 
7. Wait for the workflow to finish. (May take some time to upload depending upon video size so please have patience. Thank you.) 
8. Go to podcasters.spotify.com (Log in to your account if you haven't already).

> 9. Go to `Episodes` section. There you will see your latest upload as `Draft` 

Reviewer: is it possible to do it in a way that nobody has to login to podcasters.spotify.com but podcast is published be default as public, not draft?

Author: Sure! We can do that if we remove `SAVE_AS_DRAFT: true` line from the YAML file.

Author: resolved with ccb3c66e6ea2a570b1822009b961ba513f237af6

10. Click on the latest episode.

> 11. Update the episode details as per your requirement.

Reviewer: are these not taken from the YT video description?

Author: @derberg 
Yes, they are. But in some cases I've noticed (2 out of 10 cases) that the description was blank and nothing was added. Maybe a bug or something.
So, how can we manage those 2 out of 10 cases?

Author: So basically, the workflow (in some cases) did not pickup the description automatically and I added the description manually.

I'm thinking of doing some more test runs on my account. I will do about 10 runs and let you know the results.

Also, will remove the `save as draft` label.

Reviewer: basically best experience will be if instruction do not require anyone to go to spotify and do some manual stuff. Running workflow should be enough, and best if in logs one can find a resulting link of published podcast

Author: > basically best experience will be if instruction do not require anyone to go to spotify and do some manual stuff. Running workflow should be enough, 

Hope so.

> and best if in logs one can find a resulting link of published podcast

I don't know if there is any way to do it in the GH Action itself but it can be opened as a separate issue and someone well-versed with GH Actions can raise the PR addressing it.

Reviewer: the most important is, is `click on the Publish button.` required? can't we publish by default?

then instruction about logging into spotify and changing description would be optional instruction

Author: resolved with 045a9dd
